cve_bin_tool/parsers/go.py

```diff
@@ -0,0 +1,48 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from cve_bin_tool.parsers import Parser
+from cve_bin_tool.util import ProductInfo, ScanInfo
+
+
+class GoParser(Parser):
+    def __init__(self, cve_db, logger):
+        self.cve_db = cve_db
+        self.logger = logger
+        self.filename = ""
+
+    def find_vendor(self, product, version):
+        vendor_package_pair = self.cve_db.get_vendor_product_pairs(product)
+        vendorlist: list[ScanInfo] = []
+        if vendor_package_pair != []:
+            # To handle multiple vendors, return all combinations of product/vendor mappings
+            for v in vendor_package_pair:
+                vendor = v["vendor"]
+                file_path = self.filename
+                self.logger.debug(f"{file_path} {product} {version} by {vendor}")
+                vendorlist.append(
+                    ScanInfo(ProductInfo(vendor, product, version), file_path)
+                )
+            return vendorlist if len(vendorlist) > 0 else None
+        return None
+
+    def run_checker(self, filename):
+        self.filename = filename
+        with open(self.filename) as fh:
+            lines = fh.readlines()
+            packages = False
+            for line in lines:
+                line = line.strip()
+                if line == "require (":
+                    packages = True
+                    continue
+                if line == ")":
+                    packages = False
+                    continue
+                if packages:
+                    product = line.split(" ")[0].split("/")[-1]
+                    version = line.split(" ")[1][1:].split("-")[0].split("+")[0]
+                    vendor = self.find_vendor(product, version)
+                    if vendor is not None:
```

Comment:
Not to be solved here, but... I wonder if we should be generally doing something more fancy when find_vendor finds nothing?  I think in a lot of cases we're finding something that doesn't have any CVEs so there's nothing more we can do from a "does it have vulnerabilities?" perspective, but we probably want to improve our output to the user that says "we found this but it looks like it's never had a CVE" for their records (and if it turns out that's not true, so they can update the metadata or file a bug with us so the correct vendor is found in future).

Reply:
Sure! That would be a great future addition for further information gathering, also might lead to some interesting insights as well I suppose.